build_tools/python/e2e_test_framework/definitions/common_definitions.py

```diff
@@ -47,8 +52,8 @@ class DeviceArchitecture(_ArchitectureInfo, Enum):
   RV32_GENERIC = (ArchitectureType.CPU, "riscv_32", "generic")
 
   # Mobile GPUs
-  VALHALL_MALI = (ArchitectureType.GPU, "valhall", "mali")
-  ADRENO_GENERIC = (ArchitectureType.GPU, "adreno", "generic")
+  ADRENO_GENERIC = (ArchitectureType.GPU, "adreno", "", "qualcomm")
+  MALI_VALHALL = (ArchitectureType.GPU, "", "valhall", "arm")
```

Comment:
Originally I use the format "architecture_microarchitecture" for the constant name. So if we decide to drop `mali` and use `arm` instead, should this be `VALHALL_ARM`?

Reply:
Ah, I see your description. In general i think we always want to have the architecture name set. If `valhall` is just a microarchitecture, is there an architecture name for it?

Note that the ``architecture` here means the general ISA and `microarchitecture` is a specific hardware implementation of the ISA. For example Intel CPU has x86_64 architecture + CascadeLake microarchitecture

Reply:
Yeah, the GPU land is messy about this actually---it's far from unified like CPU architectures where we have x86_64/ARM64 to rule them all. Instead, each GPU vendor have its own way of naming stuff. Even for the same GPU vendor, different generations of GPUs can mean vastly different chip impl, which is what microarchitecture is more concerned about. Take ARM as an example, valhall is a vastly different architecture than previous generations like bifrost. The ISA corresponding to that of course changes, but GPU vendor won't tell us exactly what's the ISA and what generation it is. And there are GPU vendors like Qualcomm that basically tell us nothing about the ISA or even microarchitecture.. we just have an Adreno branding name there.. 

So I won't aim for perfection here. Just a way to make things readable.